components/widgets/settings/setting.tsx

```diff
@@ -22,18 +22,25 @@ const Setting: React.FC<Props> = ({
     footer,
     helpText,
 }: Props) => {
-    return (
-        <div
-            data-testid={inputId}
-            className='form-group'
-        >
+    let labelContent;
+    if (label) {
+        labelContent = (
             <label
                 data-testid={inputId + 'label'}
                 className={'control-label ' + labelClassName}
                 htmlFor={inputId}
             >
                 {label}
             </label>
+        );
+    }
+
+    return (
+        <div
+            data-testid={inputId}
+            className='form-group'
```

Comment:
nit: would move this as the first prop of the div (to have same ordering as in prev. file - maintaining a consistent ordering of prop types for div, etc. makes spotting a missing property easier).

Reply:
@catalintomai, I'm sorry, but don't see what exactly you are asking. Are you suggesting to swap the ordering of `className` and `data-testid`?